### PR TITLE
Closes #26966: enable wallpapers V2

### LINKED ISSUE #26966: Enable wallpapers v2 flag
The `wallpaperV2Enabled` FeatureFlag should be enabled in time for 106

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FNXV2-21559)

### PR DESCRIPTION
**Pull request by MatthewTighe**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
Fixes #26966

### CODE REVIEW

**Comment by MatthewTighe** [comment]
We should land this after https://github.com/mozilla-mobile/fenix/pull/26909

File: app/src/main/java/org/mozilla/fenix/FeatureFlags.kt
Comment (sarah541) [inline app/src/main/java/org/mozilla/fenix/FeatureFlags.kt:120]:
```suggestion
    const val wallpaperV2Enabled = true
```

**Review by sarah541 — APPROVED** [review]

**Comment by MatthewTighe** [comment]
https://github.com/mozilla-mobile/fenix/pull/26685 also *must* be landed before landing this patch

**Comment by MatthewTighe** [comment]
The dependencies for this PR have landed, but I think we should wait until after the cut to land this and then uplift it

**Comment by MatthewTighe** [comment]
@mergify backport releases_v106.0.0

**Pull request merged**
